File: CS/1313_Decompress_Run-Length_Encoded_List.cs
public class Solution {
    public int[] DecompressRLElist(int[] nums) {
        IList<int> result = new List<int>();
        for(int i = 0; i < nums.Length; i = i+2){
            int freq = nums[i];
            int val = nums[i+1];
            for(int j = 0; j < freq; j++){
                result.Add(val);
            }
        }
        return result.ToArray();
    }
}

File: Java/1313_Decompress_Run-Length_Encoded_List.java
class Solution {
    public int[] decompressRLElist(int[] nums) {
        List<Integer> result = new ArrayList<>();
        for(int i = 0; i < nums.length; i=i+2){
            int freq = nums[i];
            int val = nums[i+1];
            for(int j = 0; j < freq; j++){
                result.add(val);
            }
        }
        return result.stream().mapToInt(i->i).toArray();
    }
}
